### PR TITLE
Roku Error Codes

### PR DESCRIPTION
**Pull request by philanderson888**

### CODE REVIEW

**Comment by philanderson888** [comment]
test failing - have to amend regexMatches to use `for (const regexMatch of Object.keys(regexMatches)) {`

**Comment by philanderson888** [comment]
Hey @TwitchBronBron ready for review - I have taken the work as far as I can, just require your input before I proceed further, thanks.

**Comment by philanderson888** [comment]
cc @chrisdp

File: src/LogOutputManager.ts [hunk 1]
```diff
@@ -172,6 +266,33 @@ export class LogOutputManager {
...
+            regex = new RegExp(regex.source, 'g' +
+                (regex.ignoreCase ? 'i' : '') +
+                (regex.multiline ? 'm' : '') +
+                (regex.sticky ? 'y' : ''));
+        }
+
+        while (match === regex.exec(str)) {
+            // If you want to use regex.lastIndex in this loop, you'd need more
+            // code here to fix IE < 9
+
```
Comment (TwitchBronBron) [inline src/LogOutputManager.ts:285]:
```suggestion
```

**Review by TwitchBronBron — APPROVED** [review]

**Pull request merged**